### PR TITLE
Add popup component and status buttons

### PR DESCRIPTION
**Pull request by saralohr**
# Adds `Popup` component and some button modifiers.

## `Popup`
Includes modifiers for several popup positions:
- Default - up and centered
popup-default

- Overlay - on top of the triggering element, no pointer
popup-overlay

- Down - below the triggering element, centered by default
popup-down

- Right - right-aligned with triggering element (can be combined with down modifier)
popup-right

- Right Hang - hangs off the right side of an element. Useful for small buttons or icon buttons.
popup-right-hang

- Left - left-aligned with triggering element (can be combined with down modifier)
popup-left

- Left Hang - hangs off the left side of an element. Useful for small buttons or icon buttons.
popup-left-hang

# Status buttons

Two options for status buttons.

1. Button is neutral by default, and when hovered or `is-active` (as in ButtonGroup), it gets yes/no/maybe color

button-status-1

2. Button has yes/no/maybe color by default
button-status-2

# Small square button

Button intended for symbols or icons, has fixed width and height equal to height of `Button--small`.

screen shot 2018-04-09 at 9 59 43 am

### CODE REVIEW

File: src/css/components/Button.scss [hunk 1]
```diff
@@ -1,8 +1,6 @@
 @import "../config/index.scss";
 
-/**
- * @define default Button
- */
+// * @define default Button
```
Comment (saralohr) [inline src/css/components/Button.scss]: Just changing markup so docs aren't included in the compiled CSS.

File: src/css/components/ButtonGroup.scss [hunk 2]
```diff
@@ -18,8 +18,11 @@ $ButtonGroup-space: -1 * $border-width-small;
 // * 1. Overlap elements and override any vertical margins.
 // * 2. Reset `border-radius` so corners line up.
 
+.ButtonGroup > .Button + .Button {
+  margin-left: $ButtonGroup-space; // 1
```
Comment (saralohr) [inline src/css/components/ButtonGroup.scss]: The way this was done previously (adding negative right and left margins to all buttons in button group) was overkill. We only need it on one side of adjacent buttons to make it so the border isn't doubled up.

File: src/css/components/Button.scss [hunk 3]
```diff
@@ -166,9 +161,17 @@ $Button--small-padding: 0 $su-small;
   padding: $Button--small-padding;
 }
 
-/**
- * Modifier: Button Large
- */
+// * Modifier: Button Square small
```
Comment (williamdingwall) [inline src/css/components/Button.scss]: nice, I like this for the icon buttons

Comment (saralohr) [reply src/css/components/Button.scss]: If in the future we want default and large icon buttons, we can add those too.

**Pull request merged**